src/BSON/Iterator.c

```diff
@@ -263,10 +298,92 @@ static HashTable* php_phongo_iterator_get_properties(phongo_compat_object_handle
 	return php_phongo_iterator_get_properties_hash(object, false);
 }
 
+/* Iterator handlers */
+static void php_phongo_iterator_it_dtor(zend_object_iterator* iter)
+{
+	zval_ptr_dtor(&iter->data);
+}
+
+static int php_phongo_iterator_it_valid(zend_object_iterator* iter)
+{
+	php_phongo_iterator_t* intern = Z_ITERATOR_OBJ_P(&iter->data);
+
+	return intern->valid ? SUCCESS : FAILURE;
```

Comment:
I'm once again reminded how dumb PHP is for having `SUCCESS` defined as `0`, when bools work exactly the other way around. Took me a moment to figure out why as I was initially returning `intern->valid` directly.

Reply:
I've been bitten by this before, but I've made peace with assuming PHP's behavior is inspired by process exit codes.